papas/path.py

```diff
@@ -114,11 +126,78 @@ def distquad (time):
         minim_answer = opti.bracket(distquad, xa = -0.5e-14, xb = 0.5e-14)
         self.closest_t = minim_answer[1]
         vector_IP = self.point_at_time(minim_answer[1]) - vertex
-        self.signIP= vector_IP.Dot(self.p4.Vect().Unit())
-        self.IP = minim_answer[4]**(1.0/2)*sign(self.signIP)
-        #self.closest=TVector3(self.fx(t), self.fy(t), self.fz(t))
+        Pj = jet.p4().Vect().Unit()
+        signIP  = vector_IP.Dot(Pj)
+        self.IP = minim_answer[4]**(1.0/2)*sign(signIP)
+        
+        x,y,z = self.coord_at_time(minim_answer[1])
+        self.IPx, self.IPy, self.IPz = (x-vertex.x()),  (y-vertex.y()), (z-vertex.z())
+       
+    def compute_theta_0(self):
+        '''Computes the square root of the variance, sigma, of the multiple
+        scattering angle due to matter interactions, using the formula in PDG
+        paper. The values for the beam pipe are set'''
+        # for the beam pipe,  X_0 = 1.72e-2, width = 1.5e-3
+        P = self.p4.Vect().Dot(self.udir)
+        PT= self.p4.Perp()
+        x = abs( epaisseur_beampipe * 1.0* P/PT)
```

Comment:
not sure about this. 
the angle should be obtained from the direction of the particle at the beam pipe, not its 4 momentum.
Indeed the direction can change in the magnetic field. 
Also, it seems to me that you assume the particle is coming from 0,0,0. 
What if not the case? 
Maybe you could use the points beampipe_in and beampipe_out on the path to get the amount of material that has been crossed.